src/utils/utils.js

```diff
@@ -295,8 +295,25 @@ export const getconstructorParams = (abiConstructor, state, vals, crowdsaleNum)
     return params;
 }
 
-export function roundToDecimals(n, input) {
-  return 1.0 / n * Math.ceil(n * input)
+export const roundToDecimals = (n, input) => Math.floor10(input, n)
+
+const decimalAdjust = (type, value, exp) => {
+    if (typeof exp === 'undefined' || +exp === 0) {
+      return Math[type](value);
+    }
+    value = +value;
+    exp = +exp;
+    if (isNaN(value) || !(typeof exp === 'number' && exp % 1 === 0)) {
+      return NaN;
+    }
+    value = value.toString().split('e');
```

Comment:
Instead of these three variable assignments that are self referencing, build a util function to extract this logic and return the final value for 'value'

Reply:
Let's use  Math.floor10 and decimalAdjust. It is well-known solution: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round

Reply:
sure. Can you rename 'value' to something more descriptive?

Reply:
value -> inputNumber